lib/upload_creator.rb

```diff
@@ -262,7 +262,9 @@ def execute_convert(from, to, debug = false)
   end
 
   def should_downsize?
-    max_image_size > 0 && filesize >= max_image_size
+    max_image_size > 0 &&
+      filesize >= max_image_size &&
+      !(@image_info.type.to_s == "gif" && FastImage.animated?(@file))
```

Comment:
Do we need both tests? Wouldn't `FastImage.animated?` be enough?

Reply:
Yes, it is. `FastImage.animated?` returns `nil` for any non-GIF image.